src/network/network_server.cpp

```diff
@@ -989,6 +990,7 @@ NetworkRecvStatus ServerNetworkGameSocketHandler::Receive_CLIENT_AUTH_RESPONSE(P
 	auto authentication_method = this->authentication_handler->GetAuthenticationMethod();
 	switch (this->authentication_handler->ReceiveResponse(p)) {
 		case NetworkAuthenticationServerHandler::AUTHENTICATED:
+			this->peer_public_key = this->authentication_handler->GetPeerPublicKey();
```

Comment:
nitpick: would it be cleaner to put this a bit lower next to `receive_encryption_handler` and friends?

The other cases of the case `return` anyway :)